src/extension/providers/liquid/index.ts

```diff
@@ -0,0 +1,40 @@
+import { postMessage } from "../postMessage";
+
+declare global {
+  interface Window {
+    liquid: LiquidProvider;
+  }
+}
+
+export default class LiquidProvider {
+  enabled: boolean;
+
+  constructor() {
+    this.enabled = false;
+  }
+
+  async enable() {
+    if (this.enabled) {
+      return { enabled: true };
+    }
+    return await this.execute("enable");
```

Comment:
we do need to update the enabled variable after enabling  the provider
refer: https://github.com/getAlby/lightning-browser-extension/pull/2594